sprintf.c

```diff
@@ -519,8 +519,11 @@ rb_str_format(int argc, const VALUE *argv, VALUE fmt)
 	VALUE sym = Qnil;
 
 	for (t = p; t < end && *t != '%'; t++) ;
-	if (t + 1 == end) ++t;
 	PUSH(p, t - p);
```

Comment:
I committed in fd8df3ab3d935b4a201f9cb28598c95e4bb1ef14 with the original order as this seems an unrelated issue, and the extra exception should not affect this logic.
Please fix if you think it is worth changing.